dom/attr/attr.js

```diff
@@ -84,22 +84,28 @@ var formElements = {"INPUT": true, "TEXTAREA": true, "SELECT": true},
 			}
 		}
 	},
-	setChildOptions = function(el, value){
-		if(value != null) {
-			var child = el.firstChild,
-				hasSelected = false;
-			while(child) {
-				if(child.nodeName === "OPTION") {
-					if(value === child.value) {
-						hasSelected = child.selected = true;
-						break;
-					}
-				}
-				child = child.nextSibling;
+	_findOptionToSelect = function (parent, value) {
+		var child = parent.firstChild;
+		while (child) {
+			if (child.nodeName === 'OPTION' && value === child.value) {
+				return child;
 			}
-			if(!hasSelected) {
-				el.selectedIndex = -1;
+			if (child.nodeName === 'OPTGROUP') {
```

Comment:
are there any other valid elements?  Part of me wonders if we would also want to support custom elements that could contain `<option>` elements.